clash-prelude/src/Clash/Explicit/SimIO.hs

```diff
@@ -0,0 +1,333 @@
+{-|
+  Copyright   :  (C) 2019, Google Inc
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+
+  I\/O actions that are translatable to HDL
+-}
+
+{-# LANGUAGE DataKinds, GADTs, TypeApplications #-}
+{-# LANGUAGE BangPatterns, MagicHash, TypeOperators, ScopedTypeVariables, FlexibleContexts #-}
+
+module Clash.Explicit.SimIO
+  ( -- * I\/O environment for simulation
+    mealyIO
+  , SimIO
+  -- * Display on stdout
+  , display
+  -- * End of simulation
+  , finish
+  -- * Mutable values
+  , Reg
+  , reg
+  , readReg
+  , writeReg
+  -- * File I\/O
+  , File
+  , openFile
+  , closeFile
+  -- ** Reading and writing characters
+  , getChar
+  , putChar
+  -- ** Reading strings
+  , getLine
+  -- ** Detecting the end of input
+  , isEOF
+    -- ** Buffering operations
+  , flush
+    -- ** Repositioning handles
+  , seek
+  , rewind
+  , tell
+  )
+where
+
+import Control.Monad (when)
+import Data.Coerce
+import Data.IORef
+import GHC.TypeLits
+import Prelude hiding (getChar, putChar, getLine)
+import qualified System.IO as IO
+import System.IO.Unsafe
+
+import Clash.Annotations.Primitive (hasBlackBox)
+import Clash.Promoted.Nat
+import Clash.Signal.Internal
+import Clash.Sized.Unsigned
+import Clash.Sized.Vector (Vec (..))
+import Clash.XException (seqX)
```

Comment:
Maybe this should actually be the preferred way of formatting imports. I really dislike all the alignment we need to do.